terraform/hcl2json.go

```diff
@@ -6,9 +6,55 @@ import (
 	"github.com/hashicorp/hcl/v2/hclsyntax"
 )
 
-// ExtractVariables extract the variables from the provided file
-// When there are multiple files in a module, the variables get merged before being passed to the ParseHclToJson function
-func ExtractVariables(fileName string, fileContent string) (VariableMap, error) {
+// ParseModule iterated through all the provided files in a module
+// It extracts the variables from each one, merges them, and dereferences them one by one
+func ParseModule(files map[string]interface{}) map[string]interface{} {
+	failedFiles := make(map[string]interface{})
+	parsedFiles := make(map[string]interface{})
+
+	variablesMaps := []VariableMap{}
+	for fileName, fileContentInterface := range files {
+		// need to use interface{} for gopherjs, so cast it back to string
+		fileContent, ok := fileContentInterface.(string)
+		if !ok {
+			continue
+		}
+		variableMap, err := extractVariables(fileName, fileContent)
+		if err != nil {
+			// skip non-user errors
+			if isUserError(err) {
+				failedFiles[fileName] = err.Error()
+			}
+			continue
+		}
+		variablesMaps = append(variablesMaps, variableMap)
+	}
+
+	// merge variables so they can be used across multiple files
+	variableMap := mergeVariables(variablesMaps)
+
+	for fileName, fileContent := range files {
+		if failedFiles[fileName] == nil {
+			parsedJson, err := parseHclToJson(fileName, fileContent.(string), variableMap)
```

Comment:
noob q: isn't fileContent already a string here? or we are just casting it "just in case" again, to be safe?